toolchain/semantics/semantics_parse_tree_handler.cpp

```diff
@@ -710,24 +708,27 @@ auto SemanticsParseTreeHandler::HandleVariableDeclaration(
   auto last_child = node_stack_.pop_back_val();
   CARBON_CHECK(last_child.result_id.is_valid());
 
+  SemanticsNodeId storage_id;
   if (parse_tree_->node_kind(last_child.parse_node) !=
       ParseNodeKind::PatternBinding()) {
     SemanticsNodeId init_id = last_child.result_id;
-    auto bind_id = PopWithResult(ParseNodeKind::VariableInitializer());
+    auto storage_id = PopWithResult(ParseNodeKind::VariableInitializer());
 
     auto binding = node_stack_.pop_back_val();
     CARBON_CHECK(parse_tree_->node_kind(binding.parse_node) ==
                  ParseNodeKind::PatternBinding());
     CARBON_CHECK(binding.name_id.is_valid());
 
     // Restore the name now that the initializer is complete.
-    AddNameToLookup(binding.name_id, bind_id);
+    AddNameToLookup(binding.name_id, storage_id);
 
-    auto storage_id = semantics_->GetNode(bind_id).GetAsBindName().second;
     auto storage_type = TryTypeConversion(parse_node, storage_id, init_id,
                                           /*can_convert_lhs=*/false);
     AddNode(SemanticsNode::MakeAssign(parse_node, storage_type, storage_id,
                                       init_id));
+  } else {
+    // No init.
+    storage_id = last_child.result_id;
```

Comment:
I don't really follow this -- `storage_id` doesn't seem to be used after this... Should the `push` below use it? Or can it be removed? Or maybe I'm confused in some other way?

Reply:
Fixed, removing the variable.